include/silo/storage/column/insertion_index.h

```diff
@@ -0,0 +1,97 @@
+#ifndef SILO_INSERTION_INDEX_H
+#define SILO_INSERTION_INDEX_H
+
+#include <array>
+#include <memory>
+#include <regex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+#include <roaring/roaring.hh>
+
+#include "silo/common/nucleotide_symbol_map.h"
+#include "silo/common/nucleotide_symbols.h"
+
+namespace boost::serialization {
+struct access;
+}  // namespace boost::serialization
+
+namespace silo::storage::column::insertion {
+
+class InsertionIndex {
+   friend class boost::serialization::access;
+
+  public:
+   using three_mer_t = std::array<NUCLEOTIDE_SYMBOL, 3>;
+   using sequence_ids_t = roaring::Roaring;
+   using insertion_ids_t = std::vector<uint32_t>;
+
+  private:
+   template <class Archive>
+   [[maybe_unused]] void serialize(Archive& archive, const uint32_t /* version */) {
+      // clang-format off
+      archive & insertion_positions;
+      archive & collected_insertions;
+      // clang-format on
+   }
+
+   struct Insertion {
+      friend class boost::serialization::access;
+
+      template <class Archive>
+      [[maybe_unused]] void serialize(Archive& archive, const uint32_t /* version */) {
+         // clang-format off
+         archive & value;
+         archive & sequence_ids;
+         // clang-format on
+      }
+
+      std::string value;
+      sequence_ids_t sequence_ids;
+   };
```

Comment:
Why are those inner classes / structs? This makes the outer class quite lengthy and I don't really see a benefit of having it nested in another class.

Reply:
I moved them out now